pkg/zonemgr/zone_file_cache.go

```diff
@@ -40,10 +39,16 @@ type ZoneFileCache struct {
 	vmiRecordsMap map[string][]string
 }
 
-func NewZoneFileCache(nameServerIP string, domain string) *ZoneFileCache {
+func NewZoneFileCache(nameServerIP string, domain string, soaSerial *int) *ZoneFileCache {
```

Comment:
why *int and not just int ?

Reply:
since I want to support nil value

Reply:
just return 0, why we need nil ?
(same for all places)